components/profile-controller/controllers/profile_controller_test.go

```diff
@@ -54,9 +54,51 @@ func TestUpdateNamespaceLabels(t *testing.T) {
 		},
 	}
 	for _, test := range tests {
-		updateNamespaceLabels(test["current"])
+		defaultKubeflowNamespaceLabels = map[string]string{
+			"katib-metricscollector-injection":      "enabled",
+			"serving.kubeflow.org/inferenceservice": "enabled",
+			"pipelines.kubeflow.org/enabled":        "true",
+			"app.kubernetes.io/part-of":             "kubeflow-profile",
+		}
+		setNamespaceLabelsFromConfig(test["current"])
 		if !reflect.DeepEqual(test["expected"], test["current"]) {
-			t.Errorf("Expect:\n%v; Output:\n%v", test["current"], test["expected"])
+			t.Errorf("Expect:\n%v; Output:\n%v", test["expected"], test["current"])
 		}
 	}
 }
+
+func TestUpdateNamespaceLabels_withLabelRemoval(t *testing.T) {
```

Comment:
nit: suggest merging this with TestEnforceNamespaceLabelsFromConfig, and allow a test case to specify config yaml.
In go, we typically prefer table tests than separate test funcs.

Reply:
Updated to use table tests.